### PR TITLE
feat(autoware_planning_msgs): add autoware planning messages

### PR DESCRIPTION
**Pull request by isamu-takagi**
Signed-off-by: Takagi, Isamu <isamu.takagi@tier4.jp>

## Description

Add route set message. Related to https://github.com/autowarefoundation/autoware_msgs/issues/34.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: autoware_planning_msgs/package.xml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  <maintainer email="isamu.takagi@tier4.jp">Takagi, Isamu</maintainer>
+  <license>Apache License 2.0</license>
+
+  <buildtool_depend>ament_cmake_auto</buildtool_depend>
+
+  <build_depend>autoware_cmake</build_depend>
+  <build_depend>rosidl_default_generators</build_depend>
+
+  <depend>autoware_ad_api_msgs</depend>
+  <depend>autoware_auto_mapping_msgs</depend>
```
Comment (kenji-miyake) [inline autoware_planning_msgs/package.xml:16]: I think we should remove this dependency in autoware_msgs.

```suggestion
```

**Comment by kenji-miyake** [comment]
@mitsudome-r Could you explain the background of this PR to Fatih-san?

File: autoware_planning_msgs/package.xml (hunk 1 above)
Comment (isamu-takagi) [reply autoware_planning_msgs/package.xml:16]: Currently, SetRoute service uses autoware_auto_mapping_msgs. It should be removed when moving autoware_auto_msgs to autoware_msgs.

```
std_msgs/Header header
geometry_msgs/Pose goal
autoware_auto_mapping_msgs/HADMapSegment[] segments
```

Comment (kenji-miyake) [reply autoware_planning_msgs/package.xml:16]: Yes, I meant you should rewrite the message.

Comment (kenji-miyake) [reply autoware_planning_msgs/package.xml:16]: I think @mitsudome-r can decide how to define this message.

Comment (isamu-takagi) [reply autoware_planning_msgs/package.xml:16]: As a result of discussion with @mitsudome-r @kenji-miyake @yukkysaito , I think we now refer to autoware_auto_msg.

Comment (kenji-miyake) [reply autoware_planning_msgs/package.xml:16]: I'm sorry but I didn't recognize we'll depend on `autoware_auto_msgs` here.
If we do so, we should discuss the approach with @xmfcx.

I suggest copying necessary messages from `autoware_auto_msgs` and depend on the new message package.

Comment (isamu-takagi) [reply autoware_planning_msgs/package.xml:16]: Create the issue https://github.com/autowarefoundation/autoware_msgs/issues/34 for this.

**Comment by isamu-takagi** [comment]
@mitsudome-r @xmfcx 
Can this PR be merged soon? Or is it better to create a temporary message in [tier4_autoware_msgs](https://github.com/tier4/tier4_autoware_msgs)?

**Comment by xmfcx** [comment]
@isamu-takagi I will review it in a couple hours. Sorry for the delay.

**Review by mitsudome-r — APPROVED** [review]

**Comment by mitsudome-r** [comment]
LGTM, but I will wait for @xmfcx's comment before merging.

**Comment by isamu-takagi** [comment]
@xmfcx @mitsudome-r @kenji-miyake 
Could you please let me know if the review takes time? This PR has been blocking some PRs for a long time. The reason we didn't make this into [tier4_autoware_msgs](https://github.com/tier4/tier4_autoware_msgs) was that it would complicate the dependencies. If this PR is delayed in merging, I'll temporarily add a message there.

**Review by xmfcx — APPROVED** [review]
These are very small and compact messages, we can merge them. They are almost same as previous messages anyway. Thanks for your work.

**Pull request merged**